lib.nix

```diff
@@ -0,0 +1,37 @@
+{ lib }:
+let
+  inherit (import ./semver.nix { inherit lib; }) satisfiesSemver;
+
+  # Returns true if pythonVersion matches with the expression in pythonVersions
+  isCompatible = pythonVersion: pythonVersions: let
```

Comment:
maybe as part of the refactoring, moving the `let` keyword on its own line would be a good thing. It would put the `let` and `in` keywords on the same indent.